src/ts/background/index.ts

```diff
@@ -19,18 +20,23 @@ const messageLogger = (request: any, sender: Runtime.MessageSender) => {
 
 browser.runtime.onMessage.addListener(messageLogger);
 
-const badgeUpdater = async (request: any) => {
-  if (!request.activate) {
-    return;
+const handleSetBadge = async (request: MessageGeneric<any>) => {
+  console.log(`Handling set badge as got: ${JSON.stringify(request)}`);
+  if (request.props.isApi) {
+    await browser.browserAction.setBadgeText({ text: "API" });
+    await browser.browserAction.setBadgeBackgroundColor({
+      color: "#4688F1",
+    });
+  } else {
+    await browser.browserAction.setBadgeText({ text: "" });
   }
-  console.log(`Activating badge as got: ${JSON.stringify(request)}`);
-  await browser.browserAction.setBadgeText({ text: "API" });
-  await browser.browserAction.setBadgeBackgroundColor({ color: "#4688F1" });
 };
 
-browser.runtime.onMessage.addListener(badgeUpdater);
-
-const messageHandler = async (request: any, sender: Runtime.MessageSender) => {
+const messageHandler = async (
+  request: MessageGeneric<any>,
+  sender: Runtime.MessageSender
+) => {
+  // Still a mish mash of very strict and lazy type-checking
   if (messages.InitializeStore.matches(request)) {
```

Comment:
`InitializeStore` sounds like a function more then an object. What's the intended functionality for this?

Reply:
It means there are message builders for all message types with given enum type and type-checked props, and `InitializeStore` is a message that tells to initialize the store. But I've started to migrate away from the strongly type-checked messages as they make development really slow.